ext/sockets/sockets.c

```diff
@@ -482,7 +482,7 @@ static PHP_MINIT_FUNCTION(sockets)
 	REGISTER_LONG_CONSTANT("MSG_TRUNC",		MSG_TRUNC,		CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("MSG_PEEK",		MSG_PEEK,		CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("MSG_DONTROUTE", MSG_DONTROUTE,	CONST_CS | CONST_PERSISTENT);
-#ifdef MSG_EOR
+#if defined(MSG_EOR) && !defined(__APPLE__)
```

Comment:
google for `MSG_EOR macos`

it seems it is either complete unsupported or it needs `SEQPACKET`, but it is not used by php

if someone knows better, please advise

Reply:
We cannot change this (definitely not for any of the stable versions, but I don't see a compelling reason to change it for master either). The userland constant should be defined on macOS as well (even *if*  it wouldn't make sense there).

What we likely should change is socket_send.phpt; it seems to me that the `MSG_EOR` case only works by accident (where it works), since `socket_send()` is a wrapper for `send(2)`, and [POSIX specifies](https://pubs.opengroup.org/onlinepubs/9699919799/functions/send.html):

> MSG_EOR
> Terminates a record (if supported by the protocol).

The Linux [send manpage](https://man7.org/linux/man-pages/man2/send.2.html) clarifies:

> MSG_EOR (since Linux 2.2)
> Terminates a record (when this notion is supported, as for
 > sockets of type SOCK_SEQPACKET).

However, the socket we're using here is of type `SOCK_STREAM`.

It would be good if someone experienced with sockets could review [that test](https://github.com/php/php-src/blob/master/ext/sockets/tests/socket_send.phpt); and we should reconsider connecting to a domain which we don't control (yahoo.com).

Reply:
@devnexen `ext/sockets/tests/socket_send.phpt` test is failing on MacOS (not in CI, as online tests are skipped), do you think the code can be fixed or the `MSG_EOR` test be skipped like in https://github.com/php/php-src/pull/8390/files#diff-a516d52fc571b5da772a0ff19826e9bd371e3562eab95ca4a2b71108dda1da8bR26?